compose/src/main/kotlin/com/yourssu/handy/compose/Chip.kt

```diff
@@ -0,0 +1,111 @@
+import androidx.compose.foundation.layout.wrapContentWidth
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.yourssu.handy.compose.foundation.*
+import com.yourssu.handy.compose.icons.HandyIcons
+import com.yourssu.handy.compose.icons.line.*
+
+enum class ChipSize(
+    internal val chipHeight: Dp
+) {
+    Large(32.dp),
+    Small(24.dp)
+}
+
+enum class ChipState {
+    Unselected,
+    Selected,
+    Disabled
+}
+
+@Composable
+fun Chip(
+    text: String = "",
+    modifier: Modifier = Modifier,
+    chipState: ChipState = ChipState.Unselected,
+    chipSize: ChipSize = ChipSize.Large,
+    leadingIcon: ImageVector? = HandyIcons.Line.InfoCircle,
+    trailingIcon: ImageVector? = HandyIcons.Line.Close,
+    chipOnClick: (() -> Unit)? = null,
+    leadingIconOnClick: (() -> Unit)? = null,
+    trailingIconOnClick: (() -> Unit)? = null
+) {
+
+    val backgroundColor = remember(chipState) {
+        when (chipState) {
+            ChipState.Unselected -> ColorGray100
+            ChipState.Selected -> ColorViolet100
+            ChipState.Disabled -> ColorGray50
+        }
+    }
+
+    val textColor = remember(chipState) {
+        when (chipState) {
+            ChipState.Unselected -> ColorGray700
+            ChipState.Selected -> ColorViolet500
+            ChipState.Disabled -> ColorGray300
+        }
+    }
+
+    val textStyle: HandyTextStyle =
+        if (chipState == ChipState.Selected) HandyTypography.B3Sb14 else HandyTypography.B3Rg14
+
+    Box(
+        modifier = modifier
+            .wrapContentWidth(unbounded = true)
+            .height(chipSize.chipHeight)
+            .background(backgroundColor, shape = RoundedCornerShape(16.dp))
+            .clickable(
+                enabled = chipState != ChipState.Disabled,
+                onClick = { chipOnClick?.invoke() })
+            .padding(horizontal = 12.dp),
+        contentAlignment = Alignment.Center
+    ) {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.spacedBy(4.dp)
+        ) {
+            if (leadingIcon != null)
+                Icon(
+                    leadingIcon,
+                    tint = textColor,
+                    iconSize = IconSize.XS,
+                    modifier = Modifier.clickable(
+                        enabled = chipState != ChipState.Disabled,
+                        onClick = { leadingIconOnClick?.invoke() })
+                )
+
+            Text(
+                text = if (text.length < 10) text else text.take(9) + "...",
+                color = textColor,
+                fontSize = 14.dp,
+                style = textStyle
+            )
```

Comment:
```suggestion
            Text(
    text = text,
    color = textColor,
    style = textStyle,
    maxLines = 1, // 최대 1줄
    overflow = TextOverflow.Ellipsis // 길이 초과 시 말줄임표로
)
```
maxLines , overflow 속성 활용 시 `text = if (text.length < 10) text else text.take(9) + "...",` 처럼 직접 구현하실 필요없습니다!

이미 textStyle로 typo를 적용하셨어서 fontsize를 굳이 한번 더 적용하실 필요 없어요
HandyTextStyle 들어가보시면 이미 적용되어있습니다!

Reply:
overflow는 Modifier나 부모 layout 같은 배치 공간의 크기를 넘어설 때 말 줄임표 처리하는 것으로 알고 있는데
Figma의 Spec에는
> Chip 하나당 최대 글자 수 10자 이내(공백 포함)로 제한됩니다. 최대 글자 수를 넘을 시 말줄임표 처리됩니다.

로 작성되어 있습니다.
Typography와 TextOverflow를 찾아봤는데 이 부분을 TextOverflow를 이용해서 처리해주는 부분을 발견하지 못 했습니다!
혹시 글자수에 따른 말줄임표 기능이 TextOverflow에 있는지 알려주시면 감사하겠습니다!